### PR TITLE
SIT-2632 - Migration to Mule 4, documentation

### PR DESCRIPTION
**Pull request by viktoroooh**

### CODE REVIEW

File: src/main/mule/businessLogic.xml [hunk 1]
```diff
@@ -0,0 +1,284 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<mule xmlns:os="http://www.mulesoft.org/schema/mule/os"
+	xmlns:salesforce="http://www.mulesoft.org/schema/mule/salesforce"
+	xmlns:ee="http://www.mulesoft.org/schema/mule/ee/core"
+	xmlns:batch="http://www.mulesoft.org/schema/mule/batch" 
+	xmlns:http="http://www.mulesoft.org/schema/mule/http" xmlns="http://www.mulesoft.org/schema/mule/core" xmlns:doc="http://www.mulesoft.org/schema/mule/documentation" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="
```
Comment (JohnnyBendik) [inline src/main/mule/businessLogic.xml]: Remove http namespace

File: src/main/mule/config.xml [hunk 2]
```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<mule xmlns:os="http://www.mulesoft.org/schema/mule/os"
+	xmlns:salesforce="http://www.mulesoft.org/schema/mule/salesforce"
+	xmlns:ee="http://www.mulesoft.org/schema/mule/ee/core"
+	xmlns:batch="http://www.mulesoft.org/schema/mule/batch"
```
Comment (JohnnyBendik) [inline src/main/mule/config.xml]: Remove ee/core  and batch namespace

File: src/main/mule/endpoints.xml [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+</salesforce:query>
+		<os:store key="watermark" failOnNullValue="false" objectStore="watermarkStoreA" doc:name="Store lastQueryDateA">
+        <os:value><![CDATA[#[max(payload map $.LastModifiedDate)]]]></os:value>
+    </os:store>
+</flow>
+<flow name="sfdcQueryFlowB">
+		<os:retrieve key="watermark" objectStore="watermarkStoreB" target="lastQueryDateB" doc:name="Retrieve lastQueryDateB">
+        <os:default-value><![CDATA[${watermark.default.expression}]]></os:default-value>
+    </os:retrieve>
+    <salesforce:query config-ref="Salesforce_Sfdc_B_config" doc:name="Query Opportunity with filtering criteria from Salesforce Instance B">
```
Comment (JohnnyBendik) [inline src/main/mule/endpoints.xml]: ${sfdc.a.integration.user.id} in Query for Salesforce instance B?

File: pom.xml [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+
+	<parent>
+		<groupId>org.mule.templates</groupId>
+		<artifactId>templates-tooling-parent-pom</artifactId>
+		<version>2.1.0-SNAPSHOT</version>
+	</parent>
+
+	<groupId>org.mule.templates</groupId>
```
Comment (JohnnyBendik) [inline pom.xml]: Remove duplicate <groupId>org.mule.templates</groupId>

File: pom.xml [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+	<artifactId>template-sfdc2sfdc-opportunity-bidirectional-sync</artifactId>
+    <version>2.0.0-SNAPSHOT</version>
```
Comment (JohnnyBendik) [inline pom.xml]: Version: <version>2.1.0-SNAPSHOT</version>

File: doc/configReadme.xml [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+For instance if 10 records are fetched from origin instance, then 12 api calls will be made (1 + 10 + 1).
+
+When the outbound messaging is enabled in Salesforce and template trigger policy is push, specify Saleforce source organization eg. http://yourapp.cloudhub.io:80/?source=A as URL query parameter
+Also consider that all required fields of Opportunity in Salesforce should be added for the outbound messaging.
+]]>
+		</section>
+		<section id="default_endpoints_xml">
+		<![CDATA[
+This file should contain every inbound and outbound endpoint of your integration app. It is intented to contain the application API.
+In this particular template, this file contains a couple of poll inbound endpoints that query salesforce for updates using watermark as mentioned before.
```
Comment (JohnnyBendik) [inline doc/configReadme.xml]: couple of poll inbound endpoints?

File: doc/configReadme.xml [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+Outbound messaging is part of the workflow rule functionality in Salesforce. Workflow rules watch for specific kinds of field changes and trigger automatic Salesforce actions in this case sending opportunities as an outbound message to Mule HTTP listener,
+which will then further process this message and create Opportunity in target Salesforce organization.
+		</section>
+		<section id="considerations">
+<![CDATA[
+To make this Anypoint Template run, there are certain preconditions that must be considered. All of them deal with the preparations in both, that must be made in order for all to run smoothly. **Failling to do so could lead to unexpected behavior of the template.**
+]]>
+		</section>
+		<section id="application_configuration">
+		<![CDATA[
```
Comment (JohnnyBendik) [inline doc/configReadme.xml]: Add page.size property

File: src/main/mule/endpoints.xml (hunk 3 above)
Comment (viktoroooh) [reply src/main/mule/endpoints.xml]: changed, tested

**Pull request merged**